src/contracts/core/PaymentCoordinator.sol

```diff
@@ -161,12 +163,7 @@ contract PaymentCoordinator is
                 paymentCalculationEndTimestamp: paymentCalculationEndTimestamp
             })
         );
-        emit DistributionRootSubmitted(
-            rootIndex,
-            root,
-            paymentCalculationEndTimestamp,
-            activatedAt
-        );
+        emit DistributionRootSubmitted(rootIndex, root, paymentCalculationEndTimestamp, activatedAt);
```

Comment:
@ypatil12 we also need to index events these offchain for the payment updater

Reply:
on order for them to know the timestamp to lookup in the payments table

Reply:
We need an event to know when `GENESIS_PAYMENT_TIMESTAMP` is so we can figure out what the first range of DistributionRoot is. This is partially why I'm for adding the entire window in the DistributionRootSubmitted event. cc @8sunyuan

Reply:
Another option is to just use RPC and store for the very first update

Reply:
Another thing to think about - we are currently assuming the the window for the first range distribution root starts at `GENESIS_PAYMENT_TIMESTAMP`. That effectively means that the "0th" distribution root ends at this timestamp minus 1. This is expected, but just a call out

We may have to add an event for this if the off-chain logic becomes wonky here, but for now its fine